### PR TITLE
Split projects

### PR DESCRIPTION
**Pull request by adriendelsalle**
Description
---

- Make a clear split between projects
  - split versions between `libmamba`, `micromamba` and `mamba` to reflect a possible different maturity status
  - duplicate LICENSE and CHANGELOG, to start logging changes of projects separately
- Add `libmambapy` target for `libmamba` Python bindings
  - will make possible to depend on `libmambapy` instead of `mamba` that brings `conda` (from example in `quetz`)
  - make `mamba` dependent on `libmambapy`
- Rename CMake options to make them homogeneous
  - `BUILD_EXE` -> `BUILD_MICROMAMBA`
  - `BUILD_STATIC`/`BUILD_SHARED` -> `BUILD_LIBMAMBA` + `BUILD_STATIC`/`BUILD_SHARED`
  - `ENABLE_TESTS` -> `BUILD_LIBMAMBA_TESTS`
  - `BUILD_BINDINGS` -> `BUILD_LIBMAMBAPY`
- Simplify CMakeLists files
- Update CI workflows
- Remove `termcolor` from thirdparties, use it as a dependency

### CODE REVIEW

File: libmamba/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+    )
+
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        PUBLIC_HEADER "${LIBMAMBA_HEADERS}"
+        COMPILE_DEFINITIONS "LIBMAMBA_EXPORTS"
+        PREFIX ""
+        VERSION ${LIBMAMBA_BINARY_VERSION}
+        SOVERSION ${LIBMAMBA_BINARY_CURRENT}
```
Comment (JohanMabille) [inline libmamba/CMakeLists.txt:234]: The `SOVERSION` and `VERSION` properties should be slightly different. See [xeus](https://github.com/jupyter-xeus/xeus/blob/master/CMakeLists.txt#L332) for an example of how to set them?

File: libmamba/environment-dev.yml [hunk 2]
```diff
@@ -4,7 +4,6 @@ channels:
 dependencies:
   - cxx-compiler
   - cmake
-  - ccache
```
[the remark this answers is not in the document]
Comment (adriendelsalle) [reply libmamba/environment-dev.yml]: thx

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -356,29 +356,21 @@ jobs:
...
-          conda create -q -y -n mamba-tests vs2017_win-64 ccache python=$PYTHON_VERSION pip pybind11 libsolv libsodium termcolor-cpp libarchive "libcurl=7.76.1=*_0" nlohmann_json "cpp-filesystem>=1.5.8" conda cmake gtest gmock ninja reproc-cpp yaml-cpp cli11 pytest
-        env:
-          PYTHON_VERSION: ${{ matrix.python-version }}
+      - name: create build environment
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          environment-file: ./libmamba/environment-dev.yml
+          environment-name: build_env
       - name: build and run libmamba C++ tests
         shell: cmd
```
Comment (wolfv) [inline .github/workflows/main.yml]: need to use `cmd /C call {0}`

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -422,7 +422,7 @@ jobs:
         shell: cmd /C call {0}
         run: |
           call micromamba activate build_env
-          micromamba install ccache -c defaults
+          call micromamba install ccache -c defaults
```
Comment (wolfv) [inline .github/workflows/main.yml:425]: do we really need this when we're using the `cmd /C call {0}` syntax up there?

File: .github/workflows/main.yml (hunk 3 above)
Comment (adriendelsalle) [reply .github/workflows/main.yml]: also fixed

File: libmambapy/CMakeLists.txt [hunk 5]
```diff
@@ -2,6 +2,8 @@ find_package(pybind11 REQUIRED)
 
 include(GNUInstallDirs)
 
+add_definitions(-DGHC_WIN_DISABLE_WSTRING_STORAGE_TYPE)
```
[the remark this answers is not in the document]
Comment (adriendelsalle) [reply libmambapy/CMakeLists.txt]: did it!

File: libmamba/CMakeLists.txt (hunk 1 above)
Comment (adriendelsalle) [reply libmamba/CMakeLists.txt]: I implemented that @JohanMabille thanks, do you have any good pointer to understand better how it works?

**Pull request merged**